add server 添加机器过程中 使用 Ctrl+C 导致查看 List server 程序直接退出

Edit servers 过程的时候，我想在重新编辑的时候使用 Ctrl+C。
使用 Ctrl+C 后，从头开始编辑 server 信息，编辑完成后 查看  List server 程序退出

报错如下
```
03:46:43.240 SaveTo ▶ INFO 031 Save config to '/root/.gortal.yml'
03:46:43.242 ShowMenu ▶ DEBU 032 Show menu [%!s(*pui.MenuItem=&{Add server map[] <nil>  <nil> 0x6c25b0  false }) %!s(*pui.MenuItem=&{Edit server map[] <nil>  0x6c3c50 <nil>  false }) %!s(*pui.MenuItem=&{Delete server map[] <nil>  0x6c3c50
 <nil>  false })]
03:46:47.917 ShowMenu ▶ DEBU 033 Selected index: 3 subMenuLabel: back err: %!s(<nil>)
03:46:47.917 ShowMenu ▶ DEBU 034 Show menu [%!s(*pui.MenuItem=&{List servers map[] 0x6bb310  0x6c3830 <nil> You don't have a server yet, please go to the 'Edit Server' menu to add a server false }) %!s(*pui.MenuItem=&{Edit users map[] 0x6
bb320  0x6c2140 <nil>  false }) %!s(*pui.MenuItem=&{Edit servers map[] 0x6bb320  0x6c2140 <nil>  false }) %!s(*pui.MenuItem=&{Edit personal info map[] <nil>  0x6c2140 <nil>  false })]
03:46:48.712 ShowMenu ▶ DEBU 035 Selected index: 0 subMenuLabel: List servers err: %!s(<nil>)
03:46:48.712 ShowMenu ▶ DEBU 036 Selected: &{Label:List servers Info:map[] IsShow:0x6bb310 SubMenuTitle: GetSubMenu:0x6c3830 SelectedFunc:<nil> NoSubMenuInfo:You don't have a server yet, please go to the 'Edit Server' menu to add a server
 BackAfterSelected:false BackOptionLabel:}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6663b7]

goroutine 89 [running]:
github.com/TNK-Studio/gortal/config.(*Config).GetUserServers(0xc000012900, 0xc00012d890, 0x4)
        /opt/config/config.go:177 +0x1c7
github.com/TNK-Studio/gortal/core/pui.GetServersMenu.func1(0x0, 0xc0000d6850, 0xc000013330, 0xa20c78, 0x0, 0x0, 0x0)
        /opt/core/pui/server.go:285 +0xd0
github.com/TNK-Studio/gortal/core/pui.(*PUI).ShowMenu(0xc000106148, 0x777f40, 0x23, 0xc00000f3a0, 0x76d809, 0x4, 0xa20c78, 0x0, 0x0)
        /opt/core/pui/g.go:68 +0xc71
github.com/TNK-Studio/gortal/core/pui.(*PUI).ShowMainMenu(...)
        /opt/core/pui/g.go:107
github.com/TNK-Studio/gortal/core/jump.(*Service).Run(0xc000129f88, 0xc000013330)
        /opt/core/jump/g.go:51 +0x244
main.main.func1(0x7dd180, 0xc0000b2580)
        /opt/main.go:40 +0x96
github.com/gliderlabs/ssh.(*session).handleRequests.func1(0xc0000b2580)
        /go/pkg/mod/github.com/gliderlabs/ssh@v0.2.2/session.go:239 +0x3b
created by github.com/gliderlabs/ssh.(*session).handleRequests
        /go/pkg/mod/github.com/gliderlabs/ssh@v0.2.2/session.go:238 +0x467
```

--- a/config/config.go
+++ b/config/config.go
@@ -173,6 +173,9 @@ func (c *Config) GetUserByUsername(username string) *User {
 func (c *Config) GetUserServers(user *User) map[string]*Server {
 	servers := make(map[string]*Server, 0)
 	for serverKey, server := range *c.Servers {
+		if (*server).SSHUsers == nil {
+			return servers
+		}
 	loop:
 		for _, sshUser := range *server.SSHUsers {
 			if sshUser.AllowUsers == nil {

--- a/core/pui/menu.go
+++ b/core/pui/menu.go
@@ -110,16 +110,31 @@ func init() {
 			&MenuItem{
 				Label: "Add server",
 				SelectedFunc: func(index int, menuItem *MenuItem, sess *ssh.Session, selectedChain []*MenuItem) error {
+					var runningErr *error
+					var serverKey *string
+
+					defer func() {
+						if runningErr != nil && serverKey != nil {
+							server := (*config.Conf.Servers)[*serverKey]
+							if server != nil {
+								delete(*config.Conf.Servers, *serverKey)
+							}
+						}
+					}()
+
 					serverKey, _, err := AddServer(sess)
 					if err != nil {
+						runningErr = &err
 						return err
 					}
 					_, _, err = AddServerSSHUser(*serverKey, sess)
 					if err != nil {
+						runningErr = &err
 						return err
 					}
 					config.Conf.SaveTo(*config.ConfPath)
 					if err != nil {
+						runningErr = &err
 						return err
 					}
 					return nil

